src/utils/cmsUtils.ts

```diff
@@ -0,0 +1,11 @@
+export const stripLocaleFromUri = (uri: string): string => {
+  return uri.replace(/^\/(en|sv|fi)(?![a-z0-9])/i, '');
+};
+
+export const removeSurroundingSlashes = (path: string) => {
+  return path.replace(/^\/|\/$/g, '');
+};
+
+export const normalizeCmsUri = (uri: string) => {
+  return removeSurroundingSlashes(stripLocaleFromUri(uri));
```

Comment:
```suggestion
    return decodeURIComponent(removeSurroundingSlashes(stripLocaleFromUri(uri)));
```
Fixes the changing of languages with the Arabic (i.e. the rightmost header menu item in the production headless CMS).